engine/src/main/java/org/terasology/config/flexible/FlexibleConfig.java

```diff
@@ -58,9 +60,9 @@
      */
     boolean contains(SimpleUri id);
 
-    void setUnusedSettings(Map<SimpleUri, String> unusedSettings);
+    Map<SimpleUri, Setting> getSettings();
 
-    Map<SimpleUri, String> getUnusedSettings();
+    void save(Writer writer);
 
-    Map<SimpleUri, Setting> getActiveSettings();
+    void load(Reader reader);
```

Comment:
Are the reader/writer passed here because it is not possible to change their path once they have been instantiated?

Reply:
The configManager is responsible for deciding the path of serialization, creating a reader/writer for it and passing that to the FC.

This way, the FC does not have to concern itself with *where* the settings are being stored, and can focus only on the *how*.

Reply:
I understand that. But could the writer/reader be passed on Construction? If a path change requires re-instantiation, I guess the answer is no.